product_listing.go

```diff
@@ -0,0 +1,148 @@
+package goshopify
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+)
+
+const productListingBasePath = "product_listings"
+const productsListingResourceName = "product_listings"
+
+// ProductListingService is an interface for interfacing with the product listing endpoints
+// of the Shopify API.
+// See: https://shopify.dev/docs/admin-api/rest/reference/sales-channels/productlisting
+type ProductListingService interface {
+	List(interface{}) ([]ProductListing, error)
+	ListWithPagination(interface{}) ([]ProductListing, *Pagination, error)
+	Count(interface{}) (int, error)
+	Get(int64, interface{}) (*ProductListing, error)
+	GetProductIDs(interface{}) ([]int64, error)
+	Publish(int64) (*ProductListing, error)
+	Delete(int64) error
+}
+
+// ProductListingServiceOp handles communication with the product related methods of
+// the Shopify API.
+type ProductListingServiceOp struct {
+	client *Client
+}
+
+// ProductListing represents a Shopify product published to your sales channel app
+type ProductListing struct {
+	ID          int64           `json:"product_id,omitempty"`
+	Title       string          `json:"title,omitempty"`
+	BodyHTML    string          `json:"body_html,omitempty"`
+	Vendor      string          `json:"vendor,omitempty"`
+	ProductType string          `json:"product_type,omitempty"`
+	Handle      string          `json:"handle,omitempty"`
+	CreatedAt   *time.Time      `json:"created_at,omitempty"`
+	UpdatedAt   *time.Time      `json:"updated_at,omitempty"`
+	PublishedAt *time.Time      `json:"published_at,omitempty"`
+	Tags        string          `json:"tags,omitempty"`
+	Options     []ProductOption `json:"options,omitempty"`
+	Variants    []Variant       `json:"variants,omitempty"`
+	Images      []Image         `json:"images,omitempty"`
+}
+
+// Represents the result from the product_listings/X.json endpoint
+type ProductListingResource struct {
+	ProductListing *ProductListing `json:"product_listing"`
+}
+
+// Represents the result from the product_listings.json endpoint
+type ProductsListingsResource struct {
+	ProductListings []ProductListing `json:"product_listings"`
+}
+
+// Represents the result from the product_listings/product_ids.json endpoint
+type ProductListingIDsResource struct {
+	ProductIDs []int64 `json:"product_ids"`
+}
+
+// ProductID represents a products unique Shopify identifier
+type ProductID struct {
+	ProductID int64 `json:"product_id"`
+}
+
+// Resource which create product_listing endpoint expects in request body
+// e.g.
+// PUT /admin/api/2020-07/product_listings/921728736.json
+// {
+//   "product_listing": {
+//     "product_id": 921728736
+//   }
+// }
+type ProductListingPublishResource struct {
+	ProductListing struct {
+		ProductID
```

Comment:
Are there any benefits of including the type rather than just declare the field?
`ProductID int64 'json:"product_id"'`

Declaring a field will make initialization much easier to read

Reply:
I agree, I've removed the ProductID type in favour of the plain int64 in the latest commit